gossipd/gossipd.c

```diff
@@ -249,13 +249,15 @@ static const u8 *handle_channel_announcement_msg(struct daemon *daemon,
 {
 	const struct short_channel_id *scid;
 	const u8 *err;
+	bool our_channel;
```

Comment:
I would simply poll on the gossmap to see if our node exists, and if it has any channels.  I'm reworking all this code, so that approach is more reliable.